### PR TITLE
Assert `OpenReader` from file does not panic after closure

### LINKED ISSUE #211: make carv2 SectionReaders gracefully reject operations after closure
Right now, if they receive a read after a closure, they panic. See https://github.com/filecoin-project/lotus/issues/7111.

We made the blockstores handle closures gracefully here: https://github.com/ipld/go-car/pull/195. We should do the same for standard car IO, so that these readers behave just like any other IO in Go (e.g. file IO, network IO, etc.)

### PR DESCRIPTION
**Pull request by masih**
Write a test that asserts reader instantiated from `OpenReader` do not
panic and instead error gracefully if underlying IO is closed.

Fixes #211

### CODE REVIEW

File: v2/reader_test.go [hunk 1]
```diff
@@ -178,6 +178,27 @@ func TestReader_WithCarV2Consistency(t *testing.T) {
 	}
 }
 
+func TestOpenReader_DoesNotPanicAfterClosure(t *testing.T) {
+	subject, err := carv2.OpenReader("testdata/sample-wrapped-v2.car")
+	require.NoError(t, err)
+	dReaderBeforeClosure := subject.DataReader()
+	iReaderBeforeClosure := subject.IndexReader()
+	require.NoError(t, subject.Close())
+	dReaderAfterClosure := subject.DataReader()
```
Comment (aarshkshah1992) [inline v2/reader_test.go:187]: @masih Can you try this without using the new readers here ? As in use the `dReaderBeforeClosure ` and `iReaderBeforeClosure ` created before we closed the file.

Comment (masih) [reply v2/reader_test.go:187]: Sure

Comment (masih) [reply v2/reader_test.go]: Separated the tests [here](https://github.com/ipld/go-car/pull/212/commits/912af7dd665ee51430194a6533308a5b9257aa0c) for before and after.

**Comment by raulk** [comment]
Thanks @masih for the quick turnaround. Closing this as the issue is related to `NewReader` with a supplied mmapped region.

**Comment by raulk** [comment]
Oops, I wanted to close the _issue_, not the PR that adds the tests :D

**Comment by masih** [comment]
> Oops, I wanted to close the _issue_, not the PR that adds the tests :D

As punishment can I get a review on this PR please 🙏

**Review by willscott — APPROVED** [review]

**Pull request merged**